### PR TITLE
Normalized Azure configs

### PR DESCRIPTION
**Pull request by sam0044**
### Description

Azure api keys and endpoints were being injected in app settings repeatedly for each index so normalized them a level up in hierarchy to avoid that. Opened this Daniel can complete his changes for his ticket. I'll keep working on the rest of the stuff after this is merged.
  
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: integration-tests/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 Integration tests require the URL of the backend instance.
 
-We suggest using the .NET [Secret Manager](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0)
+We suggest using the .NET [Secret Manager]
```
Comment (dhaselhan) [inline integration-tests/README.md:13]: nitpick: Remove the Square brackets since its no longer a link

**Review by dhaselhan — DISMISSED** [review]
LGTM 👍

**Review by dhaselhan — DISMISSED** [review]
LGTM 👍

File: integration-tests/README.md [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 Integration tests require the URL of the backend instance.
 
-We suggest using the .NET [Secret Manager](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-8.0)
```
Comment (wdavidcamden) [inline integration-tests/README.md:13]: is this link dead?

Comment (sam0044) [reply integration-tests/README.md:13]: I did that before for another PR. But these links keep breaking. This was the same link that broke before so I just decided to remove it

Comment (wdavidcamden) [reply integration-tests/README.md:13]: I actually don't think the links are breaking, I think its just a brief network issue which causes the link check to fail. I'm actually leaning towards removing it as part of the PR process and more of a ongoing maintenance run. When I re-run the check in my PRs that fail, the second run usually works.
Can you put the links back in?

Comment (sam0044) [reply integration-tests/README.md:13]: Yes, I'll do that right away. Done.

**Review by dhaselhan — APPROVED** [review]
LGTM 👍

**Pull request merged**